frontend/src/pages/UserFormPage.tsx

```diff
@@ -0,0 +1,46 @@
+import UserForm from "@/components/forms/UserForm";
+import { ERROR_MESSAGES } from "@/enums/ErrorMessages";
+import { useUsersApi } from "@/hooks/useUsersApi";
+import { IUser } from "@/interfaces/IUser";
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { toast } from "sonner";
+
+const UserFormPage = () => {
+  const { id } = useParams();
+  const { getUser, adduser, updateUser } = useUsersApi();
+
+  const [user, setUser] = useState<IUser | null>(null);
+
+  const handleSaveUser = (user: IUser) => {
+    if (id) {
+      updateUser(id, user)
+        .then(() => toast.success(`משתמש עודכן בהצלחה!`))
+        .catch((err) =>
+          toast.error(ERROR_MESSAGES.GENERIC_ERROR_MESSAGE, { description: err.message })
+        );
+    } else {
+      adduser(user)
+        .then(() => toast.success(`משתמש נשמר בהצלחה!`))
+        .catch((err) =>
+          toast.error(ERROR_MESSAGES.GENERIC_ERROR_MESSAGE, { description: err.message })
+        );
+    }
+  };
+
+  useEffect(() => {
+    if (!id) return;
+    getUser(id)
+      .then((res) => setUser(res))
+      .catch((err) => console.log(err));
+  }, []);
+
+  return (
+    <div>
+      <h1 className="font-bold text-2xl">פרטי משתמש</h1>
+      <UserForm existingUser={user} saveInfo={(user) => handleSaveUser(user)} />
```

Comment:
render this only when the request is finished being made. if you found a user then set the user. if you didnt then it will just load without a user and then you can get rid of the use effect in the form.